zenoh_cpp_vendor/zenohc_q_declaration.md

```diff
@@ -0,0 +1,153 @@
+
+### Version Stability [1.ii]
+
+TODO(zettascale): Explain version Stability
+
+### Public API Declaration [1.iii]
+
+As a C library, elements available in [zenoh.h](https://github.com/eclipse-zenoh/zenoh-c/blob/main/include/zenoh.h) are considered to be the library's public API.
+
+### API Stability Policy [1.iv]
+
+TODO(zettascale): Explain API Stability
+
+### ABI Stability Policy [1.v]
+
+TODO(zettascale): Explain ABI Stability
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+
+TODO(zettascale): Explain ABI and ABI Stability  Within a Released ROS Distribution
+
+## Change Control Process [2]
+
+### Change Requests [2.i]
+
+Checking through the commits history, it can be seen is not the case.
+
+### Contributor Origin [2.ii]
+
+Does not have it (or it does not seem like it’s the case).
+
+### Peer Review Policy [2.iii]
+
+Seems to be followed for pull requests on the GitHub repository, but as not all code changes occur through change requests, this can not be confirmed for these changes.
+
+### Continuous Integration [2.iv]
+
+`zenoh-c` is compiled in the ROS 2 buildfarm for all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers).
+
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/zenoh_cpp_vendor/)
+* [linux_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/zenoh_cpp_vendor/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/zenoh_cpp_vendor/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/zenoh_cpp_vendor/)
+
+### Documentation Policy [2.v]
+
+Not available.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+Provided [doxygen documentation](https://zenoh-c.readthedocs.io/en) for the whole project.
+
+### Public API Documentation [3.ii]
+
+Yes, doxygen documentation is available for library [here](https://zenoh-c.readthedocs.io/en).
+
+### License [3.iii]
+
+Apache 2.0 license declared for the repository, it can be found [here](https://github.com/eclipse-zenoh/zenoh-c/blob/main/LICENSE).
+
+### Copyright Statements [3.iv]
+
+Is not available.
+
+### Quality Declaration [3.v]
+
+This document represents the Quality Declaration document for the `zenoh-c` ROS dependency.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+Tests provided to cover the expected usage of the library, for the version of the library used can be found [here](https://github.com/eclipse-zenoh/zenoh-c?tab=readme-ov-file).
+
+### Public API Testing [4.ii]
+
+Not clear without coverage results to check if all the API is covered.
+
+### Coverage [4.iii]
+
+Code coverage and internal policies are not public, if any.
+
+### Performance [4.iv]
+
+Performance tests are defined in the vendored package.
+
+### Linters and Static Analysis [4.v]
+
+Not available publicly, if any.
```

Comment:
`zenoh` uses on [rust-clippy](https://github.com/rust-lang/rust-clippy) for linting and static analysis.